apps/application-system/form/src/routes/AssignApplication.tsx

```diff
@@ -1,77 +1,98 @@
 import React, { useEffect } from 'react'
 import { useHistory, useLocation } from 'react-router-dom'
-import { useMutation } from '@apollo/client'
+import { ServerError, useMutation } from '@apollo/client'
+import { GraphQLError } from 'graphql'
+import * as Sentry from '@sentry/react'
 import qs from 'qs'
 
-import { Text, Page, Box, LoadingDots, Stack } from '@island.is/island-ui/core'
-import { NotFound } from '@island.is/application/ui-shell'
+import { ErrorShell, LoadingShell } from '@island.is/application/ui-shell'
 import { ASSIGN_APPLICATION } from '@island.is/application/graphql'
-import { getSlugFromType } from '@island.is/application/core'
+import { getSlugFromType, coreErrorMessages } from '@island.is/application/core'
+
+const parseGraphQLError = (
+  error?: GraphQLError,
+): (ServerError & Record<string, unknown>) | null => {
+  if (!error) {
+    return null
+  }
+
+  try {
+    return JSON.parse(error.message)
+  } catch {
+    return null
+  }
+}
 
 export const AssignApplication = () => {
   const location = useLocation()
   const history = useHistory()
   const queryParams = qs.parse(location.search, { ignoreQueryPrefix: true })
+  const isMissingToken = !queryParams.token
+  const [assignApplication, { loading, error }] = useMutation(
+    ASSIGN_APPLICATION,
+    {
+      onCompleted({ assignApplication }) {
+        const { id, typeId } = assignApplication
 
-  const [
-    assignApplication,
-    { loading, error: assignApplicationError },
-  ] = useMutation(ASSIGN_APPLICATION, {
-    onCompleted({ assignApplication }) {
-      const { id, typeId } = assignApplication
-
-      // fall back to application if for some reason we can not find the configuration
-      const slug = getSlugFromType(typeId) || 'application'
+        // fall back to application if for some reason we can not find the configuration
+        const slug = getSlugFromType(typeId) || 'application'
 
-      history.push(`../${slug}/${id}`)
+        history.push(`../${slug}/${id}`)
+      },
     },
-  })
-
-  const isMissingToken = !queryParams.token
-  const couldNotAssignApplication = !!assignApplicationError
+  )
 
   useEffect(() => {
-    if (isMissingToken) {
-      return
-    }
+    const init = async () => {
+      if (isMissingToken) {
+        Sentry.captureException(
+          new Error(
+            `Missing token, cannot assign the application ${location.search}`,
+          ),
+        )
```

Comment:
This can be a user error which can add to non-fixable noise in Sentry.

User errors can be logged as a warning somewhere to help with debugging. Though not sure if/how we want to log it in the browser though.

Reply:
I replaced it with a `captureMessage` instead of an exception. It might still be noisy but would help the debugging. I don't see other alternatives at the moment than going with Sentry for logging this